osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/security/ReplaceKeyCommandExecutor.java

```diff
@@ -101,58 +107,66 @@ public ActionResponseDto executeBundleAction(final DlmsConnectionManager conn, f
         LOGGER.info("Keys set on device :{}", device.getDeviceIdentification());
 
         SetKeysRequestDto setKeysRequestDto = (SetKeysRequestDto) actionRequestDto;
+
         if (!setKeysRequestDto.isGeneratedKeys()) {
+            // using AES
+            //decrypt using RSA and encrypt using AES
             setKeysRequestDto = this.reEncryptKeys((SetKeysRequestDto) actionRequestDto);
-        }
+        } //else
+        //if isGeneratedKeys() == true, then:
+        // generated keys are encrypted using AES by the GenerateAndReplaceKeyCommandExecutor
 
         final DlmsDevice devicePostSave = this.execute(conn, device, ReplaceKeyCommandExecutor
                 .wrap(setKeysRequestDto.getAuthenticationKey(), KeyId.AUTHENTICATION_KEY,
-                        SecurityKeyType.E_METER_AUTHENTICATION));
+                        SecurityKeyType.E_METER_AUTHENTICATION, setKeysRequestDto.isGeneratedKeys()));
 
         this.execute(conn, devicePostSave, ReplaceKeyCommandExecutor
                 .wrap(setKeysRequestDto.getEncryptionKey(), KeyId.GLOBAL_UNICAST_ENCRYPTION_KEY,
-                        SecurityKeyType.E_METER_ENCRYPTION));
+                        SecurityKeyType.E_METER_ENCRYPTION, setKeysRequestDto.isGeneratedKeys()));
 
         return new ActionResponseDto(REPLACE_KEYS + device.getDeviceIdentification() + WAS_SUCCESFULL);
     }
 
     private SetKeysRequestDto reEncryptKeys(final SetKeysRequestDto setKeysRequestDto) throws FunctionalException {
         final byte[] reEncryptedAuthenticationKey = this.securityKeyService
-                .reEncryptKey(setKeysRequestDto.getAuthenticationKey(), SecurityKeyType.E_METER_AUTHENTICATION);
+                .reEncryptKey(setKeysRequestDto.getAuthenticationKey()); //, SecurityKeyType.E_METER_AUTHENTICATION);
         final byte[] reEncryptedEncryptionKey = this.securityKeyService
-                .reEncryptKey(setKeysRequestDto.getEncryptionKey(), SecurityKeyType.E_METER_ENCRYPTION);
+                .reEncryptKey(setKeysRequestDto.getEncryptionKey()); //, SecurityKeyType.E_METER_ENCRYPTION);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)